package j10_exception;

//** try ~ catch ~ finally
//=> finally 블럭은 무조건 시행

public class Ex03_Finally {

	public static void main(String[] args) {

		int[] price = { 100, 200, 300 };

		// 1) 정상 실행
		for (int i = 0; i < price.length; i++) {

			try {
				System.out.printf("** price[%d] = %d\n", i, price[i]);
			} catch (Exception e) {
				System.out.println("** Exception => " + e.toString());
			} finally {
				System.out.println("** finally_무조건 실행, i => **" + i);
			}

			System.out.println("** for_endLine, i => " + i + "\n");
		} // for

		System.out.println("** Program 정상 종료 **");

		System.out.println("====================================");
//		===================================================
		// 2) Exception

		for (int i = 0; i <= price.length; i++) {

			try {
				System.out.printf("** price[%d] = %d\n", i, price[i]);
			} catch (Exception e) {
				System.out.println("** Exception => " + e.toString());
			} finally {
				System.out.println("** finally_무조건 실행, i => **" + i);
			}

			System.out.println("** for_endLine, i => " + i + "\n");
		} // for

		System.out.println("** Program 정상 종료 **");
	}// main

}
// class